cpp/box/Box.h

```diff
@@ -380,6 +380,75 @@ class Box
         });
     }
 
+    //! Calculate distance between two points using boundary conditions
+    /*! \param r_i Position of first point
+        \param r_j Position of second point
+    */
+    float computeDistance(const vec3<float>& r_i, const vec3<float>& r_j) const
+    {
+            vec3<float> r_ij = wrap(r_j - r_i);
+            return std::sqrt(dot(r_ij, r_ij));
+    }
+
+    //! Calculate distance between a set of points and query points using boundary conditions.
+     /*!\param query_points Particle position to query.
+        \param points Particle positions.
+        \param query_point_indices Array of indices that correspond to a set of query points.
+        \param point_indices Array of indices that correspond to a set of points.
+        \param distances Distances between points and query_points.
+        \param n_query_points The number of query points.
+        \param n_points The number of points.
+        \param n_query_indices The number of query points indices.
+        \param n_point_indices The number of points indices.
+    */
+    void computeDistances(const vec3<float>* query_points, const vec3<float>* points,
+        const unsigned int* query_point_indices, const unsigned int* point_indices,
+        float *distances, const unsigned int n_query_points, const unsigned int n_points,
+        const unsigned int n_query_indices, const unsigned int n_point_indices) const
```

Comment:
Minor nitpick, but we usually order this so that the array lengths go right after the arrays, e.g. `n_query_points` comes right after `query_points`.